packages/core/src/snapshot.js

```diff
@@ -323,6 +324,70 @@ function waitForDiscoveryNetworkIdle(page, options) {
 // Used to cache resources across core instances
 const RESOURCE_CACHE_KEY = Symbol('resource-cache');
 
+function* triggerResourceRequests(page, snapshot) {
+  // copy widths to prevent mutation later
+  let widths = snapshot.widths.slice();
+  let initialWidth = widths.shift();
+
+  // set the initial page size
+  yield page.resize({
+    width: initialWidth,
+    height: snapshot.minHeight
+  });
+
+  // navigate to the url
+  yield page.goto(snapshot.url);
+
+  if (snapshot.execute) {
+    // when any execute options are provided, inject snapshot options
+    /* istanbul ignore next: cannot detect coverage of injected code */
+    yield page.eval((_, s) => (window.__PERCY__.snapshot = s), snapshot);
+    yield page.evaluate(snapshot.execute.afterNavigation);
+  }
+
+  // resize and set pixel ratio for higher density images
+  if (snapshot.deviceScaleFactor !== 1) {
+    yield waitForDiscoveryNetworkIdle(page, snapshot.discovery);
+    yield page.evaluate(snapshot.execute?.beforeResize);
```

Comment:
Yay or nay: should we maybe add `execute.beforeMobile` to replace this part? Technically the width is staying the same so can it count as resizing?